### PR TITLE
Added option to set the method name for Basic authentication

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Added option to set the method name for Basic authentication

### CODE REVIEW

File: lib/auth/AuthBasic.cc [hunk 1]
```diff
@@ -96,11 +113,17 @@ AuthenticationPtr AuthBasic::create(ParamMap& params) {
     if (passwordIt == params.end()) {
         throw std::runtime_error("No password provided for basic provider");
     }
-
-    return create(usernameIt->second, passwordIt->second);
+    auto methodIt = params.find("method");
```
Comment (BewareMyPower) [inline lib/auth/AuthBasic.cc:116]: Could you add a test to cover this part that a param string is accepted? Like https://github.com/apache/pulsar-client-cpp/blob/829618d686c11217922c70b4e5ffba97559c7056/tests/AuthBasicTest.cc#L125

Comment (merlimat) [reply lib/auth/AuthBasic.cc:116]: 👍  done

File: lib/auth/AuthBasic.cc [hunk 2]
```diff
@@ -41,6 +41,14 @@ std::string base64_encode(const std::string& s) {
 AuthDataBasic::AuthDataBasic(const std::string& username, const std::string& password) {
     commandAuthToken_ = username + ":" + password;
     httpAuthToken_ = base64_encode(commandAuthToken_);
+    methodName_ = DEFAULT_BASIC_METHOD_NAME;
+}
```
Comment (BewareMyPower) [inline lib/auth/AuthBasic.cc:45]:
```suggestion
AuthDataBasic::AuthDataBasic(const std::string& username, const std::string& password)
    : AuthDataBasic(username, password, DEFAULT_BASIC_METHOD_NAME) {}
```

Reuse the other constructor.

**Review by BewareMyPower — APPROVED** [review]
Overall LGTM. Just left a minor comment.

**Comment by nodece** [comment]
Do we need to extend the `c_Authentication.cc`?

**Comment by merlimat** [comment]
> Do we need to extend the c_Authentication.cc?

Yes, we can add it later. Just want to get this included in the release and python too.

**Pull request merged**